src/main/java/com/fasterxml/jackson/databind/cfg/JsonNodeFeature.java

```diff
@@ -61,8 +61,18 @@ public enum JsonNodeFeature implements DatatypeFeature
      *
      * @since 2.15
      */
-    STRIP_TRAILING_BIGDECIMAL_ZEROES(true)
-    ;
+    STRIP_TRAILING_BIGDECIMAL_ZEROES(true),
+
+    /**
+     *
+     * Feature that determines whether to fail on attempt to coercing {@code NaN} into {@link java.math.BigDecimal}
+     * because {@link com.fasterxml.jackson.databind.DeserializationFeature#USE_BIG_DECIMAL_FOR_FLOATS} is enabled.
+     *<p>
+     * Default value is {@code true} for backwards-compatibility, but will be changed to {@code false} in 3.0.
+     *
+     * @since 2.16
+     */
+    ALLOW_NaN_TO_BIG_DECIMAL_COERCION(true);
```

Comment:
nit: Should be all-capital case.

Reply:
Come to think of it, I am not super happy with name here... not sure what'd be better but should either emphasize failure (FAIL_ON_NAN_TO_BIG_DECIMAL_COERCION) or skip/ignore...

Reply:
Fail makes sense, because that's what happens 👍🏼